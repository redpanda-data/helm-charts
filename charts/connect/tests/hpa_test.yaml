suite: hpa tests
templates:
  - hpa.yaml
tests:
  - it: should enable autoscaler default
    set:
      autoscaling:
        enabled: true
    asserts:
      - matchSnapshot:
          path: spec

  - it: should configure autoscaler custom
    set:
      autoscaling:
        enabled: true
        minReplicas: 2
        maxReplicas: 28
        metrics:
          - type: Resource
            resource:
              name: cpu
              target:
                type: Utilization
                averageUtilization: 20
          - type: Resource
            resource:
              name: memory
              target:
                type: Utilization
                averageUtilization: 20
        behavior:
          scaleDown:
            stabilizationWindowSeconds: 200
            policies:
              - type: Pods
                value: 3
                periodSeconds: 40
    asserts:
      - matchSnapshot:
          path: spec
