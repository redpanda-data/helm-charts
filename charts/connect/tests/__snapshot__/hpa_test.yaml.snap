should configure autoscaler custom:
  1: |
    behavior:
      scaleDown:
        policies:
          - periodSeconds: 40
            type: Pods
            value: 3
        stabilizationWindowSeconds: 200
    maxReplicas: 28
    metrics:
      - resource:
          name: cpu
          target:
            averageUtilization: 20
            type: Utilization
        type: Resource
      - resource:
          name: memory
          target:
            averageUtilization: 20
            type: Utilization
        type: Resource
    minReplicas: 2
    scaleTargetRef:
      apiVersion: apps/v1
      kind: Deployment
      name: RELEASE-NAME-redpanda-connect
should enable autoscaler default:
  1: |
    maxReplicas: 12
    metrics:
      - resource:
          name: cpu
          target:
            averageUtilization: 80
            type: Utilization
        type: Resource
    minReplicas: 1
    scaleTargetRef:
      apiVersion: apps/v1
      kind: Deployment
      name: RELEASE-NAME-redpanda-connect
